mbed_devices/_internal/windows/disk_aggregation.py

```diff
@@ -0,0 +1,189 @@
+"""Data aggregation about a disk on Windows."""
+from abc import ABC, abstractmethod
+from typing import List, Generator, Optional
+from typing import NamedTuple
+
+from mbed_devices._internal.windows.component_descriptor import ComponentDescriptor, UNKNOWN_VALUE
+from mbed_devices._internal.windows.disk_drive import DiskDrive
+from mbed_devices._internal.windows.disk_partition import DiskPartition
+from mbed_devices._internal.windows.disk_partition_logical_disk_relationships import (
+    DiskPartitionLogicalDiskRelationship,
+)
+from mbed_devices._internal.windows.logical_disk import LogicalDisk
+from mbed_devices._internal.windows.volume_set import VolumeInformation, get_volume_information
+
+
+class DataAggregation(NamedTuple):
+    """Data aggregation with regards to a disk."""
+
+    Label: str  # e.g. C:
+    Description: str  # e.g. Removal Disk
+    FreeSpace: int
+    Size: int
+    PartitionName: str  # e.g. Disk #0, Partition #2
+    PartitionType: str  # 16-bit FAT
+    VolumeInformation: VolumeInformation
+    Caption: str  # e.g. SAMSUNG MZNLN512HMJP-000L7
+    PhysicalDiskName: str
+    Model: str
+    InterfaceType: str  # e.g. IDE
+    MediaType: str  # e.g. Fixed hard disk media
+    Manufacturer: str
+    SerialNumber: str
+    Status: str
+    PNPDeviceID: str
+
+
+class DiskDataAggregation(ComponentDescriptor, DataAggregation):
+    """Disk information based on lots of different sources."""
+
+    @property
+    def class_name(self) -> str:
+        """Returns the class name."""
+        return "DiskDataAggregation"
+
+    @property
+    def device_id(self) -> str:
+        """Returns the device id field."""
+        return self.Label
+
+    @classmethod
+    def iter_all(cls) -> Generator["ComponentDescriptor", None, None]:
+        """Gets a generator over all elements currently registered in the system."""
+        yield from ()
+
+
+class DataAggregator(ABC):
```

Comment:
This name seems a bit ambiguous to me, what kind of data are we aggregating? The docstring mentions "disk data", what is that? I suggest using more descriptive names and providing more context in the docstrings.